l1-contracts/src/core/interfaces/messagebridge/IFrontier.sol

```diff
@@ -3,7 +3,22 @@
 pragma solidity >=0.8.18;
 
 interface IFrontier {
-  function insertLeaf(bytes32 _leaf) external;
+  /**
+   * @notice Inserts a new leaf into the frontier tree and returns its index
+   * @param _leaf - The leaf to insert
+   * @return The index of the leaf in the tree
+   */
+  function insertLeaf(bytes32 _leaf) external returns (uint256);
 
+  /**
+   * @notice Returns the root of the frontier tree
+   * @return The root of the tree
+   */
   function root() external view returns (bytes32);
+
+  /**
+   * @notice Returns whether the tree is full
+   * @return Whether the tree is full
```

Comment:
Me being pedantic. But would prefer `true if full, false otherwise` just to make sure that there is no way it can be misunderstood.